packages/core/helper-plugin/lib/src/components/GenericInput/index.js

```diff
@@ -292,7 +298,13 @@ const GenericInput = ({
           id={name}
           hint={hint}
           name={name}
-          onChange={onChange}
+          onChange={event => {
+            if (event.target.value === '') {
```

Comment:
Honestly I don't think it is a good fix here, you are trying to provide a fix for you specific use case in the CM that will impact all the other plugins.

Reply:
@soupette What would you suggest? I don't like the fix either, but I don't have a better idea and the file contains some fixes of that type already ...

Reply:
I think we need to implement an `action` on the event propagated by the design system input:

```
onChane={({ target: { type: 'my-type', value: '', action: 'clear'}) => ....
```

React Select does something similar, and then it's up to the developer to deal with the received event.

WDYT?

Reply:
Mh, that's an interesting way of looking at it. I'm wondering what this would change in terms of this PR though because I suppose the check would then become

```ls
if (event.target.action === 'clear') {
  // ...
}
```

What do you think of giving this PR a go and starting a discussion on the event in the FE sync? I feel like we need to look at it as a whole, because right now we do have the situation:

- API sends null
- FE converts to undefined or empty string (because the DS can not handle undefined in all cases and null is forbidden by react)
- in case of an empty string we need to send back null

It is a lot of type-casting and the event proposal doesn't really fix it either yet. Your proposal makes the API clearer though and could be a good moment to start discussing a unification of DS component APIs.

Reply:
@gu-stav I discussed this topic with @soupette during a 1:1 
the proposition is to move this logic to the `handleChange` we can find in the `EditViewDataManagerProvider` mainly because updating a component from the helper-plugin is risky + the CTB needs to make the difference between undefined/null/empty string
it fixed this issue temporarily inside the CM:
- without risking any breaking change for helper-plugin + CTB
- giving us time to bring this conversation during FE sync

I'm ready to push my new commit, WDYT?

Reply:
Mh, interesting - didn't know there is another dumpster with fixes. Yeah, I think I'm fine with that. Maybe we should move some of the existing onChange fixes, such as 

https://github.com/strapi/strapi/blob/b6b500b75355f38e0e8401257bd57e48a642149c/packages/core/helper-plugin/lib/src/components/GenericInput/index.js#L254

there too?

Reply:
@gu-stav I added select and number empty string value checks inside the `EditViewDataManagerProvider`'s `handleChange` as well!

before possibly merging this fix I would like to validate that `date` type doesn't seem to need this check - I verified and it receives null as initialValue then null onClear action, I removed it - let me know if I missed something @soupette

Reply:
> Mh, that's an interesting way of looking at it. I'm wondering what this would change in terms of this PR though because I suppose the check would then become
> 
> ```livescript
> if (event.target.action === 'clear') {
>   // ...
> }
> ```
> 
> What do you think of giving this PR a go and starting a discussion on the event in the FE sync? I feel like we need to look at it as a whole, because right now we do have the situation:
> 
> * API sends null
> * FE converts to undefined or empty string (because the DS can not handle undefined in all cases and null is forbidden by react)
> * in case of an empty string we need to send back null
> 
> It is a lot of type-casting and the event proposal doesn't really fix it either yet. Your proposal makes the API clearer though (by hiding the complexity) and could be a good moment to start discussing a unification of DS component APIs.

I won't give this PR a go you are providing a fix in the helper plugin for the content manager without changing the codebase of the CTB or other plugins that are using this component. What you should do instead is providing a fix in the content manager only and not the helper plugin.

Reply:
hola @soupette I actually removed all the changes from the helper plugin to [implement it directly into the CM ](https://github.com/strapi/strapi/pull/13313/commits/aff13a0d60dbe6969d9c55e1badf0b4540dce684)! that's why I asked for another review from you 😊